### ISSUE TITLE
important_packages argparse breaks CI

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#16169]
important_packages argparse breaks CI

### Current Output
```
  FAIL: argparse C
  Test "argparse" in category "nimble-packages-1"
  Failure: reBuildFailed
  package test failed
  $ nimble test
    Verifying dependencies for argparse@1.0.0
    Compiling /Users/runner/work/Nim/Nim/pkgstemp/argparse/tests/test1 (from package argparse) using c backend
  stack trace: (most recent call last)
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/src/argparse.nim(93, 12) domkParser
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/src/argparse/backend.nim(760, 27) generateDefs
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/src/argparse/backend.nim(689, 19) helpProcDef
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/src/argparse/backend.nim(685, 9) getHelpText
  /Users/runner/work/Nim/Nim/lib/pure/strutils.nim(2873, 11) stripLineEnd
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/tests/test1.nim(25, 7) template/generic instantiation of `suite` from here
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/tests/test1.nim(26, 8) template/generic instantiation of `test` from here
  /Users/runner/work/Nim/Nim/pkgstemp/argparse/tests/test1.nim(27, 22) template/generic instantiation of `newParser` from here
  /Users/runner/work/Nim/Nim/lib/pure/strutils.nim(2873, 11) Error: attempt to access a nil address kind: rkInt
         Tip: 1 messages have been suppressed, use --verbose to show them.
       Error: Execution failed with exit code 256
          ... Command: /Users/runner/work/Nim/Nim/bin/nim c --noNimblePath -d:NimblePkgVersion=1.0.0 --hints:off -r --path:. /Users/runner/work/Nim/Nim/pkgstemp/argparse/tests/test1
  
  PASS: arraymancer C                                                (35.84 sec)

```

### Additional Information

in unrelated PR https://github.com/nim-lang/Nim/pull/16166/checks?check_run_id=1466652591

**Comment by ringabout** [comment]
This is caused by:
https://github.com/nim-lang/Nim/issues/15939 
Every `^1` usage for string in VM may be affected.

**Comment by timotheecour** [comment]
right, i just noticed too: here's git bisect output:
```
527e792827497002ded742ad815c567a4a056471 is the first bad commit
commit 527e792827497002ded742ad815c567a4a056471
Author: hlaaftana <10591326+hlaaftana@users.noreply.github.com>
Date:   Thu Nov 12 13:44:21 2020 +0300

    make var string return var char w/ BackwardsIndex (#15461)

    * make var string return var char w/ BackwardsIndex

    fixes #14497

    * work around VM bug
    * properly workaround again

 lib/pure/pathnorm.nim    | 6 +++---
 lib/system.nim           | 1 +
 tests/stdlib/tstring.nim | 7 +++++++
 3 files changed, 11 insertions(+), 3 deletions(-)
bisect run success
```

my PR https://github.com/nim-lang/Nim/pull/15987 fixes that
